constraints/cpu.md

```diff
@@ -157,6 +157,83 @@ docker run --cpus 4 -ti debian bash -c "nproc; cat /sys/fs/cgroup/cpu,cpuacct/cp
 400000  # cfs quota
 ```
 
+
+## CPU cgroups
+
+Kata Containers run over two layers of cgroups, the first layer is in the guest where
+only the workload is placed, the second layer is in the host that is more complex and
+might contain more than one process and task (thread) depending of the number of
+containers per POD and vCPUs per container. Following diagram represents a nginx container
+created with docker with the default number of vcpus.
+
+
+```
+docker run -dt --runtime=kata-runtime nginx
+
+
+       .-------.
+       | nginx |
+    .--'-------'---.  .------------.
+    | Guest Cgroup |  | Kata agent |
+  .-'--------------'--'------------'.    .-----------.
+  |  Thread: Hypervisor's vCPU 0    |    | Kata Shim |
+ .'---------------------------------'.  .'-----------'.
+ |             Tasks                 |  |  Processes  |
+.'-----------------------------------'--'-------------'.
+|                    Host Cgroup                       |
+'------------------------------------------------------'
+```
+
+In next sections is explained what is the difference between processes and tasks and why
+only hypervisor's vCPUs are constrained.
+
+### cpu cgroups in the guess
+
+Only the workload process including all its threads are placed into cpu cgroups, this means
+that kata agent and systemd run without constraints in the guest.
+
+#### CPU pinning (cpuset cgroup)
+
+Kata Containers try to apply and honor the cgroups but sometimes that's not possible, this
```

Comment:
s/try/tries/. I'd split this into two sentences for easier reading. Something like:

> Kata Containers tries to apply and honor the cgroups but sometimes that is not possible. An example of this occurs with cpu cgroups when the number of virtual CPUs (in the guest) does not match the actual number of physical host CPUs.